BlueprintUI/Sources/Element/Element.swift

```diff
@@ -64,3 +64,37 @@ public protocol Element {
     func backingViewDescription(bounds: CGRect, subtreeExtent: CGRect?) -> ViewDescription?
 
 }
+
+
+//
+// MARK: Value Type Validation
+//
+
+
+///
+/// This protocol exists to enforce at compile time that your `Element` are value types like `struct` or `enum`.
+///
+/// It is very very unusual and usually an error to make an `Element` a `class` type. Blueprint's internal
+/// implementation relies on the fact that passed in `Element` respect value semantics and are owned by the framework.
+///
+/// Notes
+/// -----
+/// You should really not make your `Element` be a `class`. If for some reason you really really want to do this,
+/// (you should not do this unless you have a good reason, which you do not), then override the
+/// `elements_should_be_value_types_by_overriding_this_method_i_acknowledge_i_am_in_hard_mode()`
+/// method in your type to opt out of this validation. But you probably shouldn't.
+///
+public protocol Blueprint_Elements_Should_Be_Value_Types {
+    func elements_should_be_value_types_by_overriding_this_method_i_acknowledge_i_am_in_hard_mode()
+}
+
+
+public extension Blueprint_Elements_Should_Be_Value_Types {
+    func elements_should_be_value_types_by_overriding_this_method_i_acknowledge_i_am_in_hard_mode() {}
+}
+
+
+public extension Blueprint_Elements_Should_Be_Value_Types where Self : AnyObject {
+    @available(*, unavailable)
```

Comment:
You can add a message to this to make the error message better:

```suggestion
    @available(*, unavailable, message: "Elements should be structs")
```

Reply:
Will do! Will commit locally since I'm already adding the underscore.